lib/web_catalog/api_extractor.es6.js

```diff
@@ -140,511 +242,440 @@ foam.CLASS({
+
+      // Add to ctorId: New APIs that:
+      // (1) Are not exposed by some interface lower level than protoId's, AND
+      // (2) Are not already on ctorId.
+      const newAPIs = this.dedupFilter_(existingAPIs,
+                                       this.getClassAPIs_(protoId));
+      this.objectGraphApiMapper.map(ctorId, newAPIs, protoId);
+      this.apis_[ctorId] = ctorAPIs.concat(newAPIs);
+    },
+    function getAllProtosAPIs_(id) {
+      const og = this.objectGraph;
+      let lowerProtoId = og.getPrototype(id);
+      let apis = [];
+      while (!og.isType(lowerProtoId)) {
+        apis = this.dedupConcat_(apis, this.getClassAPIs_(lowerProtoId));
+        lowerProtoId = og.getPrototype(lowerProtoId);
+      }
+      return apis;
+    },
+    function storeAPIsFromArray_(ctorId, arr) {
+      const og = this.objectGraph;
+      let apis = this.apis_[ctorId] = this.apis_[ctorId] || [];
+      this.apis_[ctorId] = this.dedupConcat_(apis, arr);
+    },
+    function storeProto_(protoId, ctorId) {
+      const protoIdStr = protoId.toString();
+      let ctorIds = this.protos_[protoIdStr] =
+          this.protos_.hasOwnProperty(protoIdStr) ?
+          this.protos_[protoIdStr] : [];
+      if (!ctorIds.includes(ctorId)) ctorIds.push(ctorId);
+    },
+    function getClassAPIs_(id) {
+      const og = this.objectGraph;
+      if (id === this.objectDotPrototype_) {
+        // Object.prototype: Store "marked-as-built-in" APIs with "+apiName+".
+        return og.getObjectKeys(id)
+            .filter(name => name !== 'prototype')
+            .filter(this.filterConstantAPIs_.bind(this, id))
+            .map(name => {
+              const match = name.match(/^[+](.*)[+]$/);
+              if (match === null) return name;
+              return match[1];
+            });
+      }
+      if (id === this.function_ || id === this.functionDotPrototype_) {
+        // Function.prototype: Include non-writable APIs and "prototype" API.
+        return og.getObjectKeys(id).filter(name => !/^[+].*[+]$/.test(name));
+      }
+      // Default: Exclude:
+      // (1) object-graph'ified properties "+reservedName+",
+      // (2) integers,
+      // (3) the "prototype" property,
+      // (4) uninteresting constants.
+      return og.getObjectKeys(id)
+          .filter(name => !/^[+].*[+]$/.test(name))
+          .filter(name => !/^[0-9]+$/.test(name))
+          .filter(name => name !== 'prototype')
+          .filter(this.filterConstantAPIs_.bind(this, id));
+    },
+    function getInstanceAPIs_(id) {
+      const og = this.objectGraph;
+      // Exclude object-graph'ified properties "+reservedName+" and integers.
+      return og.getObjectKeys(id)
+          .filter(name => !/^[+].*[+]$/.test(name))
+          .filter(name => !/^[0-9]+$/.test(name));
+    },
+    function filterConstantAPIs_(id, name) {
+      const og = this.objectGraph;
+      if (this.constantTypes.length === 0) return true;
```

Comment:
Is this needed? I guess it saves two OG calls, but the check at the end should work the same without this.

Reply:
It's not functionally necessary. I added it as an optimization akin to the old `retainConstantMembers = false` case.